File: ruby/src/plandefinition.rb
require './fhir_client'

class PlanDefinitionClient
  def get_plandefinition(id)
    FhirClient.new

    FHIR::PlanDefinition.read(id)
  end

  def search_plandefinition
    FhirClient.new

    params = {
      'definition': 'Questionnaire/866683f3-c41b-47c0-b42f-86f9ff978d1d',
      'publisher': 'Program Creator'
    }
    FHIR::PlanDefinition.search(params)
  end
end

File: javascript/js/plan-definition.js
import Client from "./client";

class PlanDefinitionClient {

  async get() {
    const client = new Client();
    return client.request("PlanDefinition/4944e73f-e447-49ba-a64c-a246b9ef4bdd");
  }

  async search() {
    const client = new Client();
    return client.request("PlanDefinition?definition=Questionnaire/866683f3-c41b-47c0-b42f-86f9ff978d1d&publisher=Program Creator");
  }
}

export default PlanDefinitionClient;
